htlcswitch/interfaces.go

```diff
@@ -231,6 +231,9 @@ type InterceptedPacket struct {
 	// CustomRecords are user-defined records in the custom type range that
 	// were included in the payload.
 	CustomRecords record.CustomSet
+
+	// OnionBlob is the onion packet for the next hop
+	OnionBlob [lnwire.OnionPacketSize]byte
```

Comment:
Wouldn't you rather just have the fully decrypted routing information rather than the blob itself? As with the blob, then you'll need to handle properly de-deriving the shared secret and processing the blob _outside_ of lnd.

Reply:
I don't follow. This blob is the one meant for the next hop, LND can't decrypt it since it doesn't have those keys.